torch_xla/csrc/tensor_util.cpp

```diff
@@ -904,6 +904,70 @@ std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
   }
 }
 
+std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
+    const std::vector<at::Tensor>& tensors,
+    const std::vector<XLATensor::ShardingSpecPtr>& shardings,
+    const std::vector<std::string>& devices) {
+  XLA_TIMED("TensorToData");
+  XLA_CHECK_EQ(tensors.size(), shardings.size());
+  XLA_CHECK_EQ(tensors.size(), devices.size());
+
+  std::vector<xla::ComputationClient::DataPtr> handles;
+  for (size_t i = 0; i < tensors.size(); ++i) {
+    torch::lazy::BackendDevice device = ParseDeviceString(devices[i]);
+    xla::Shape shape = CreateComputationShapeFromTensor(tensors[i], &device);
+
+    std::vector<xla::ComputationClient::TensorSource> source_tensors;  // in
+    std::vector<xla::ComputationClient::DataPtr> new_handles;          // out
+    if (shardings[i] != nullptr) {
+      xla::OpSharding sharding = shardings[i]->sharding;
+      // TODO(yeounoh) PJRT runs a process per host and without cross host
```

Comment:
Can you clarify in this comment that you're talking about the SPMD case? Without SPMD, we still run 4 processes per host by default

Reply:
Ack.

Reply:
Done.